### PR TITLE
Merge blockchain generator modules and test them

### PR DESCRIPTION
**Pull request by geigerzaehler**
Merge, simplify, and document the modules `Oscoin.Test.Crypto.Blockchain.Generator` and `Oscoin.Test.Crypto.Blockchain.Arbitrary`. Also add tests for the blockchain generators.

### CODE REVIEW

**Comment by cloudhead** [comment]
Didn't we want to keep generators and arbitrary separate? Don't remember.

**Comment by geigerzaehler** [comment]
> Didn't we want to keep generators and arbitrary separate? Don't remember.

This probably applied to keeping `Arbitrary` orphan instances separate. But here we just have functions that return `Gen`s. This just puts them into one place and uses consistent naming.

File: test/Test/Oscoin/Crypto/Blockchain/GeneratorsTest.hs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import           Oscoin.Consensus
+import qualified Oscoin.Consensus.Nakamoto as Nakamoto
+import           Oscoin.Consensus.Validation
+import           Oscoin.Crypto.Blockchain
+import           Oscoin.Test.Crypto
+import           Oscoin.Test.Crypto.Blockchain.Generators
+import qualified Oscoin.Time.Chrono as Chrono
+
+import           Test.Tasty
+-- import           Test.Tasty.QuickCheck
```
Comment (cloudhead) [inline test/Test/Oscoin/Crypto/Blockchain/GeneratorsTest.hs]: Delete?

Comment (geigerzaehler) [reply test/Test/Oscoin/Crypto/Blockchain/GeneratorsTest.hs]: oops

**Pull request merged**